_posts/2024-06-14-solarkraft-part1.md

```diff
@@ -0,0 +1,146 @@
+
+Here's how it works:
+
+1. **Monitor Specification:** Developers define a _runtime monitor_ that captures the expected behavior of the contract using a formal language like [TLA+][]. These specifications outline
+    - **pre-conditions**: conditions that must be met before a function can execute,
+    
+    - **post-conditions**: the expected ledger state changes after each successful invocation, and
+    
+    - **failure conditions**: conditions under which the smart contract must revert.
+
+(Solarkraft introduces two novel ideas: small _modular_ monitor specifications, and a _hybrid_ monitoring approach. We will cover both in a later posts.)
+
+![]({{ site.baseurl }}/img/solarkraft-architecture.jpg)
+
+{:start="2"}
+2. **Monitoring in Action:** The Solarkraft _fetcher_ component of the runtime monitor continuously observes the blockchain for contract invocations and retrieves the transaction data.
+
+3. **Verification & Catching Deviations**: The Solarkraft _verifier_ compares the on-chain transaction data to the expected behavior defined in the monitor specification. This is an off-chain component that runs the [Apalache model-checker][Apalache] (running it on-chain would be too expensive and infeasible).
+
+4. **Alerts:** If the verifier detects any deviation from the specified behavior (e.g., an unauthorized transaction attempt or an unexpected state change), it can take predefined actions, such as pausing the smart contract or raising an on- or off-chain alert to developers.
+
+By continuously monitoring behavior against formal specifications, runtime monitoring offers several advantages over traditional methods:
+
+- **Proactive Approach:** It catches bugs in nearly real-time, limiting or preventing potential financial losses and disruptions before they occur.
+
+- **Continuous Monitoring:** Unlike testing and audits, which occur at a specific point in time against a fixed commit, runtime verification provides ongoing security throughout the contract's lifecycle.
+
+- **Early Detection:** Runtime monitoring can identify potential issues early in the execution process, allowing for quicker mitigation and reducing the attack window for malicious actors.
+
+In the following posts of this series, we'll delve deeper into the world of runtime monitoring in Solarkraft with TLA+. We'll explore how to specify contract behavior using this language, dive into Solarkraft's unique specification and monitoring strategies, and see how runtime verification can be seamlessly integrated into the development workflow to build more secure and reliable smart contracts.
+
+_Development of Solarkraft was supported by the [Stellar Development Foundation][] with a generous Activation Award from the [Stellar Community Fund][] of 50,000 USD in XLM._
+
+
+[Apalache]: https://konnov.phd/portfolio/apalache/
+[Solarkraft]: https://thpani.net/solarkraft/
+[Audits]: https://thpani.net/audits/
+[Contact]: https://thpani.net/#contact
```

Comment:
idk if you want these to link to your site or mine. feel free to change :)

Reply:
I think it's fine, if the inbounds go to you this time :)